securesystemslib/dsse.py

```diff
@@ -58,18 +61,23 @@ def from_dict(cls, data: dict) -> "Envelope":
         payload = b64dec(data["payload"])
         payload_type = data["payloadType"]
 
-        signatures = []
+        signatures = {}
         for signature in data["signatures"]:
             signature["sig"] = b64dec(signature["sig"]).hex()
-            signatures.append(Signature.from_dict(signature))
+            signature = Signature.from_dict(signature)
+            if signature.keyid in signatures:
+                raise ValueError(
+                    f"Multiple signatures found for keyid {signature.keyid}"
+                )
+            signatures[signature.keyid] = signature
 
         return cls(payload, payload_type, signatures)
 
     def to_dict(self) -> dict:
         """Returns the JSON-serializable dictionary representation of self."""
 
         signatures = []
-        for signature in self.signatures:
+        for signature in list(self.signatures.values()):
```

Comment:
Is there a reason to convert to `list`? You could iterate directly over  `dict_values`. Same comment applies below.

Reply:
ah no - I just followed the format here [comment](https://github.com/theupdateframework/python-tuf/pull/2436#discussion_r1482773245) . Will change to just iterating over dict_values

Reply:
Yeah, that PR has a few such conversions for type checking reasons.